server/handlers/main.js

```diff
@@ -0,0 +1,34 @@
+import fs from 'fs'
+import { URL } from 'node:url'
+
+import { sendResponse, sendResponseError } from '../lib/send-response.js'
+
+const responseHeaders = {
+  'Content-Type': 'text/html',
+  'Cache-Control': 'public, max-age=300, must-revalidate'
+}
+
+let responseCache = null
+
+export default async function handleMain(req, res) {
+  let filePath = new URL('../../client/dist/index.html', import.meta.url)
+
+  if (responseCache) {
+    sendResponse(res, 200, responseHeaders, responseCache)
+  } else {
+    fs.access(filePath, fs.constants.F_OK, error => {
+      if (error) {
+        sendResponseError(res, 404, 'Not Found')
```

Comment:
We could get `404` and `500` together since you are running `fs.access` and `fs.readFile` in parallel.